parlai/agents/starspace_torch/starspace_torch.py

```diff
@@ -0,0 +1,367 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+#
+# Simple implementation of the starspace algorithm, slightly adapted for dialogue.
+# See: https://arxiv.org/abs/1709.03856
+
+from parlai.core.dict import DictionaryAgent
+from parlai.core.torch_ranker_agent import TorchRankerAgent
+from parlai.core.torch_agent import Output
+from .modules import Starspace
+
+import torch
+from torch import optim
+import torch.nn as nn
+import os
+import random
+
+
+class StarspaceTorchAgent(TorchRankerAgent):
+    """Simple implementation of the starspace algorithm: https://arxiv.org/abs/1709.03856
+    """
+
+    OPTIM_OPTS = {
+        'adadelta': optim.Adadelta,
+        'adagrad': optim.Adagrad,
+        'adam': optim.Adam,
+        'adamax': optim.Adamax,
+        'asgd': optim.ASGD,
+        'lbfgs': optim.LBFGS,
+        'rmsprop': optim.RMSprop,
+        'rprop': optim.Rprop,
+        'sgd': optim.SGD,
+    }
+
+    @staticmethod
+    def dictionary_class():
+        return DictionaryAgent
+
+    @staticmethod
+    def add_cmdline_args(argparser):
+        """Add command-line arguments specifically for this agent."""
+        TorchRankerAgent.add_cmdline_args(argparser)
+        agent = argparser.add_argument_group('StarSpace Arguments')
+        agent.add_argument(
+            '-esz', '--embeddingsize', type=int, default=128,
+            help='size of the token embeddings')
+        agent.add_argument(
+            '-enorm', '--embeddingnorm', type=float, default=10,
+            help='max norm of word embeddings')
+        agent.add_argument(
+            '-shareEmb', '--share-embeddings', type='bool', default=True,
+            help='whether LHS and RHS share embeddings')
+        agent.add_argument(
+            '--lins', default=0, type=int,
+            help='If set to 1, add a linear layer between lhs and rhs.')
+        agent.add_argument(
+            '-lr', '--learningrate', type=float, default=0.1,
```

Comment:
Double check what's covered in TA already:
https://github.com/facebookresearch/ParlAI/blob/9e113105d1690deda0b372f6aeaf933ab58e3a72/parlai/core/torch_agent.py#L161

Reply:
i added learning rate as an arg here to change the default to 1 from 0.1 (in TA)